### PR TITLE
[Automation] Enhance patch_igz script

### PR DESCRIPTION
**Pull request by liranbg**
1. DRY
2. Added patching alerts service
3. Fix resetting DB

https://iguazio.atlassian.net/browse/ML-8572

### CODE REVIEW

File: automation/patch_igz/patch_remote.py [hunk 1]
```diff
@@ -268,113 +280,80 @@ def _push_docker_images(self, built_images):
...
-                "default-tenant",
-                "patch",
-                "deployment",
-                "mlrun-api-worker",
-                "-p",
-                f"{self._deploy_patch}",
-            ]
-        )
+        for deployment in self._deployments:
+            logger.info(f"Patching deployment {deployment}")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:284]:
```suggestion
            logger.info(f"Patching deployment", deployment=deployment)
```

File: automation/patch_igz/patch_remote.py [hunk 2]
```diff
@@ -268,113 +280,80 @@ def _push_docker_images(self, built_images):
...
-                "-n",
-                "default-tenant",
-                "set",
-                "image",
-                "deployment/mlrun-api-worker",
-                f"{container}={built_image}",
-            ]
-        )
+        for deployment in self._deployments:
+            logger.info(f"Replace container {container} for {deployment}")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:312]:
```suggestion
            logger.info(f"Replace container in deployment", container=container, deployment=deployment)
```

File: automation/patch_igz/patch_remote.py [hunk 3]
```diff
@@ -268,113 +280,80 @@ def _push_docker_images(self, built_images):
...
-                "default-tenant",
-                "rollout",
-                "restart",
-                "deployment",
-                "mlrun-api-chief",
-                "mlrun-api-worker",
-            ]
-        )
+        for deployment in self._deployments:
+            logger.info(f"Restarting deployment {deployment}")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:327]:
```suggestion
            logger.info(f"Restarting deployment", deployment=deployment)
```

File: automation/patch_igz/patch_remote.py [hunk 4]
```diff
@@ -268,113 +280,80 @@ def _push_docker_images(self, built_images):
...
-                "status",
-                "deployment",
-                "mlrun-api-worker",
-                "--timeout=120s",
-            ],
-            live=True,
-        )
-        logger.info("Waiting for mlrun-api to become ready")
+        for deployment in self._deployments:
+            logger.info(f"Waiting for {deployment} to become ready")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:341]:
```suggestion
            logger.info(f"Waiting for deployment to become ready", deployment=deployment)
```

File: automation/patch_igz/patch_remote.py [hunk 5]
```diff
@@ -386,90 +365,72 @@ def _wait_deployment_ready(self):
...
+        current_non_mlrun_db_services = {
+            deployment: int(replicas)
+            for deployment, replicas in dict(
+                output.split(":")
+                for output in current_non_mlrun_db_services_output.split()
+            ).items()
+        }
 
-        logger.info("Scaling down mlrun-api-chief")
+        logger.info("Scaling down services down")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py]:
```suggestion
        logger.info("Scaling down services")
```

File: automation/patch_igz/patch_remote.py [hunk 6]
```diff
@@ -386,90 +365,72 @@ def _wait_deployment_ready(self):
...
                 "wait",
-                "pods",
-                "-l",
-                "app.kubernetes.io/sub-component=chief",
                 "--for=delete",
-                "--timeout=60s",
+                "--timeout=300s",
+                "pod",
+                "--selector",
+                "app.kubernetes.io/component!=db,app.kubernetes.io/name=mlrun",
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py]:
```suggestion
                non_mlrun_db_services_deployment_selector,
```

File: automation/patch_igz/patch_remote.py [hunk 7]
```diff
@@ -386,90 +365,72 @@ def _wait_deployment_ready(self):
                 "app.kubernetes.io/name=mlrun",
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:365]: Why are we also waiting for mlrun-db and mlrun-ui to be ready?

File: automation/patch_igz/patch_remote.py [hunk 8]
```diff
@@ -386,90 +365,72 @@ def _wait_deployment_ready(self):
...
-                "--timeout=240s",
+                "--timeout=300s",
             ],
             live=True,
         )
 
     def _reset_mlrun_db(self):
-        curr_worker_replicas = (
+        non_mlrun_db_services_deployment_selector = (
+            "app.kubernetes.io/component!=db,app.kubernetes.io/name=mlrun"
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py]: Why is mlrun-ui also being scaled down?

File: automation/patch_igz/patch_remote.py [hunk 9]
```diff
@@ -478,56 +439,34 @@ def _reset_mlrun_db(self):
...
-
-        for line in self._exec_remote(cmd).splitlines()[1:]:
-            if "mlrun-db" in line:
-                return line.split()[0]
-
-    @staticmethod
-    def _get_image_tag(tag) -> str:
-        return f"{tag}"
+        for deployment, replicas in current_non_mlrun_db_services.items():
+            logger.info(f"Scaling up {deployment} with {replicas}")
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py:460]:
```suggestion
            logger.info(f"Scaling up deployment", deployment=deployment, replicas=replicas)
```

**Review by moranbental — CHANGES_REQUESTED** [review]

File: automation/patch_igz/patch_remote.py (hunk 1 above)
Comment (liranbg) [reply automation/patch_igz/patch_remote.py:284]: this script has no structured logger but python basic one

File: automation/patch_igz/patch_remote.py (hunk 7 above)
Comment (liranbg) [reply automation/patch_igz/patch_remote.py:365]: ensure application is patched without side effects

File: automation/patch_igz/patch_remote.py (hunk 8 above)
Comment (liranbg) [reply automation/patch_igz/patch_remote.py]: hmm, I will fix that

File: automation/patch_igz/patch_remote.py [hunk 10]
```diff
@@ -386,90 +365,74 @@ def _wait_deployment_ready(self):
...
                 "condition=Ready",
-                "--timeout=240s",
+                "--timeout=300s",
             ],
             live=True,
         )
 
     def _reset_mlrun_db(self):
-        curr_worker_replicas = (
+        non_mlrun_db_services_deployment_selector = (
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py]: now the `non_mlrun_db_services_deployment_selector` name  is incorrect 
Maybe - `non_mlrun_deployment_selector`?

File: automation/patch_igz/patch_remote.py [hunk 11]
```diff
@@ -386,90 +365,74 @@ def _wait_deployment_ready(self):
...
+            "app.kubernetes.io/component!=ui,"
+            "app.kubernetes.io/component!=db,"
+            "app.kubernetes.io/name=mlrun"
+        )
+
+        # in form of "deployment1:replicas1\ndeployment2:replicas2\n..."
+        current_non_mlrun_db_services_output = (
```
Comment (moranbental) [inline automation/patch_igz/patch_remote.py]: same here

**Review by moranbental — CHANGES_REQUESTED** [review]

File: automation/patch_igz/patch_remote.py (hunk 10 above)
Comment (liranbg) [reply automation/patch_igz/patch_remote.py]: will use `mlrun_api_services_deployment_selector`

**Pull request merged**